pkgs/os-specific/darwin/libtapi/default.nix

```diff
@@ -18,24 +18,33 @@ stdenv.mkDerivation {
       sha256 = "1a19h39a48agvnmal99n9j1fjadiqwib7hfzmn342wmgh9z3vk0g";
     };
 
+  sourceRoot = "source/src/llvm";
+
   nativeBuildInputs = [ cmake python3 ];
 
-  buildInputs = [ clang_6.cc ];
+  # ncurses is required here to avoid a reference to bootstrap-tools, which is
+  # not allowed for the stdenv.
+  buildInputs = [ ncurses ];
+
+  cmakeFlags = [ "-DLLVM_INCLUDE_TESTS=OFF" ];
 
+  # fixes: fatal error: 'clang/Basic/Diagnostic.h' file not found
+  # adapted from upstream
+  # https://github.com/tpoechtrager/apple-libtapi/blob/3cb307764cc5f1856c8a23bbdf3eb49dfc6bea48/build.sh#L58-L60
   preConfigure = ''
-    cd src/llvm
-  '';
+    INCLUDE_FIX="-I $PWD/projects/clang/include"
+    INCLUDE_FIX+=" -I $PWD/build/projects/clang/include"
 
-  cmakeFlags = [ "-DLLVM_INCLUDE_TESTS=OFF" ];
+    cmakeFlagsArray+=(-DCMAKE_CXX_FLAGS="$INCLUDE_FIX")
+  '';
 
-  buildFlags = [ "libtapi" ];
+  buildFlags = [ "clangBasic" "libtapi" ];
 
-  installTargets = [ "install-libtapi" "install-tapi-headers"];
+  installTargets = [ "install-libtapi" "install-tapi-headers" ];
 
   meta = with lib; {
     license = licenses.apsl20;
     maintainers = with maintainers; [ matthewbauer ];
     broken = stdenv.hostPlatform == stdenv.targetPlatform;
```

Comment:
@Ericson2314 This looks related to reflex-platform cross-compilation, but it causes CI to fail.  Is this an expected (and accepted) failure?  Is there a way to coax this PR and those downstream through CI so that iterative improvements can be made asynchronously?